### PR TITLE
Feature/kr/migration polymesh

### PR DESCRIPTION
**Pull request by KanishkaRajputd**

### CODE REVIEW

File: pages/api/v1/getVotesForPolymeshPIPs/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { isDataExist } from '../posts/on-chain-post';
+import apiErrorWithStatusCode from '~src/util/apiErrorWithStatusCode';
+import { ProposalType } from '~src/global/proposalType';
+
+const checkProposalType = (proposalType: ProposalType) => {
+	const isValidProposalType = ([ProposalType.TECHNICAL_PIPS, ProposalType.UPGRADE_PIPS, ProposalType.COMMUNITY_PIPS].includes(proposalType));
+	return isValidProposalType;
+};
+
+const handler: NextApiHandler<{ data: any } | { error: string | null }> = async (req, res) => {
```
Comment (alphainfinitus) [inline pages/api/v1/getVotesForPolymeshPIPs/index.ts:19]: please add types here. Use existing and standardised types used in other apis

File: pages/api/v1/getVotesForPolymeshPIPs/index.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		return res.status(400).json({ data: null, error: 'Invalid network in request header' });
+	}
+	const numPostId = Number(postId);
+	if (isNaN(numPostId) || numPostId < 0) {
+		throw apiErrorWithStatusCode(`The postId "${postId}" is invalid.`, 400);
+	}
+
+	const strProposalType = String(proposalType) as ProposalType;
+	if (!isProposalTypeValid(strProposalType) && checkProposalType(proposalType)) {
+		throw apiErrorWithStatusCode(`The proposal type "${proposalType}" is invalid.`, 400);
```
Comment (alphainfinitus) [inline pages/api/v1/getVotesForPolymeshPIPs/index.ts:34]: message from the messages file please

File: pages/api/v1/getVotesForPolymeshPIPs/index.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	const variables = {
+		index_eq: postId,
```
Comment (alphainfinitus) [inline pages/api/v1/getVotesForPolymeshPIPs/index.ts:37]: please check if this should be `postId` or `numPostId`

File: src/util/getSubstrateAddress.ts [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
+import { encodeAddress } from '@polkadot/util-crypto';
+
+/**
+ * Return an address encoded for the current network
+ *
+ * @param address An address
+ *
+ */
+export default function getSubstrateAddress(address: string): string | null {
+	if(!address) return null;
```
Comment (alphainfinitus) [inline src/util/getSubstrateAddress.ts:14]: this is not necessary, the function does not have an optional argument

File: pages/api/v1/latest-activity/on-chain-posts.ts [hunk 5]
```diff
@@ -64,12 +63,13 @@ export async function getLatestActivityOnChainPosts(params: IGetLatestActivityOn
 		if (network === 'collectives') {
 			query = GET_PROPOSALS_LISTING_BY_TYPE_FOR_COLLECTIVES;
 		}
-
-		let subsquidRes: any = {};
+		if(network === 'polymesh'){
+			query = GET_PROPOSALS_LISTING_FOR_POLYMESH;
+		}let subsquidRes: any = {};
```
Comment (alphainfinitus) [inline pages/api/v1/latest-activity/on-chain-posts.ts:68]: new line missing between if block and `subsquidRes`

File: pages/api/v1/posts/on-chain-post.ts [hunk 6]
```diff
@@ -79,6 +78,13 @@ export interface IReactions {
 	};
 }
 
+export interface IPIPsVoting {
+	balance: null | string;
```
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:82]: instead of storing null, keep balance and voter optional

File: src/components/Listing/index.tsx [hunk 7]
```diff
@@ -1,123 +1,122 @@
...
+
+const Listing: FC<IListingProps> = (props) => {
+	const { className, proposalType, isTip, tipStartedIndex } = props;
+
+	const { network } = useNetworkContext();
+
+	const [posts, setPosts] = useState(props.posts || []);
+
+	useEffect(() => {
+		setPosts(props.posts || []);
```
Comment (alphainfinitus) [inline src/components/Listing/index.tsx:30]: this seems unnecessary

File: src/global/post_categories.ts [hunk 8]
```diff
@@ -1,23 +1,26 @@
...
+	PROPOSAL = 'proposal',
+	MOTION = 'motion',
+	TREASURY_PROPOSAL = 'treasury proposal',
+	TECH_COMMITTEE_PROPOSAL = 'tech committee proposal',
+	BOUNTY = 'bounty',
+	CHILD_BOUNTY = 'child bounty',
+	TIP = 'tip',
+	ALLIANCE_MOTION = 'alliance motion',
+	ANNOUNCEMENT = 'alliance announcement',
+	TECHNICAL_PIPS = 'technical_PIPs',
```
Comment (alphainfinitus) [inline src/global/post_categories.ts:18]: please double check if  the values need to be snake case (`_`) and plural

File: src/global/proposalType.ts [hunk 9]
```diff
@@ -1,262 +1,292 @@
...
+	| 'Bounty'
+	| 'Tip'
+	| 'ChildBounty'
+	| 'ReferendumV2'
+	| 'FellowshipReferendum'
+	| 'TechnicalCommittee'
+	| 'Community'
+	| 'UpgradeCommittee';
+
+export function getSubsquidProposalType(proposalType: Exclude<ProposalType, ProposalType.DISCUSSIONS | ProposalType.GRANTS>): TSubsquidProposalType | undefined{
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:58]: this shouldn't return `undefined`

File: src/ui-components/VotesProgressInListing.tsx [hunk 10]
```diff
@@ -44,8 +44,10 @@ const VotesProgressInListing = ({ tally, index, onchainId,status, proposalType,
 		return Number(formatBnBalance(bn, { numberAfterComma: 6, withThousandDelimitor: false }, network));
 	};
 
-	const ayeVotesNumber = ( proposalType === getSubsquidProposalType(ProposalType.FELLOWSHIP_REFERENDUMS)) ? (ayes) : bnToIntBalance(tallyData.ayes || ZERO);
-	const totalVotesNumber = ( proposalType === getSubsquidProposalType(ProposalType.FELLOWSHIP_REFERENDUMS)) ? (ayes + nays) : bnToIntBalance(tallyData.ayes?.add(tallyData.nays|| ZERO) || ZERO);
+	const check = [getSubsquidProposalType(ProposalType.FELLOWSHIP_REFERENDUMS), ProposalType.TECHNICAL_PIPS, ProposalType.UPGRADE_PIPS].includes(proposalType as TSubsquidProposalType );
```
Comment (alphainfinitus) [inline src/ui-components/VotesProgressInListing.tsx:47]: better variable name please

File: src/queries.ts [hunk 11]
```diff
@@ -1,1274 +1,1386 @@
...
+  }
+  proposalsConnection(orderBy: id_ASC, where: {type_eq: $type_in}) {
+    totalCount
+  }
+}
+
+`;
+export const GET_PROPOSALS_LISTING_FOR_POLYMESH =`
+query PolymeshPrposalsQuery($type_in: ProposalType, $limit: Int = 10, $offset: Int = 0) {
+  proposals(orderBy: createdAt_DESC, limit: $limit, offset: $offset, where: {type_eq: $type_in}) {
```
Comment (alphainfinitus) [inline src/queries.ts:140]: variable `type_in` is assigned to `type_eq`

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Pleaes check comments.

File: pages/api/v1/posts/on-chain-post.ts (hunk 6 above)
Comment (KanishkaRajputd) [reply pages/api/v1/posts/on-chain-post.ts:82]: y value subsquid se hi null aari rhi h (balance: null)

File: src/queries.ts [hunk 12]
```diff
@@ -1,1274 +1,1386 @@
...
+      block
+      status
+      timestamp
+    }
+  }
+}
+`;
+
+export const GET_PROPOSALS_LISTING_BY_TYPE = `
+query PolymeshPrposalsQuery($type_in: ProposalType, $limit: Int = 10, $offset: Int = 0,$orderBy: [ProposalOrderByInput!] = createdAtBlock_DESC,) {
```
Comment (alphainfinitus) [inline src/queries.ts:108]: type_in should accept array for back compatibility

File: src/queries.ts [hunk 13]
```diff
@@ -1,1274 +1,1386 @@
...
-      block
-      status
-      timestamp
-    }
-  }
-}
-`;
-
-export const GET_PROPOSALS_LISTING_BY_TYPE = `
-query ProposalsListingByType($type_in: [ProposalType!], $orderBy: [ProposalOrderByInput!] = createdAtBlock_DESC, $limit: Int = 10, $offset: Int = 0, $index_in: [Int!], $hash_in: [String!], $trackNumber_in: [Int!], $status_in: [ProposalStatus!]) {
```
Comment (alphainfinitus) [inline src/queries.ts:109]: ProposalsListingByType looks to be replaced incorrectly

**Review by alphainfinitus — APPROVED** [review]

File: src/global/proposalType.ts [hunk 14]
```diff
@@ -1,262 +1,292 @@
...
+	BOUNTIES = 'bounties',
+	TIPS = 'tips',
+	CHILD_BOUNTIES = 'child_bounties',
+	OPEN_GOV = 'referendums_v2',
+	REFERENDUM_V2 = 'referendums_v2',
+	DISCUSSIONS = 'discussions',
+	GRANTS = 'grants',
+	ANNOUNCEMENT = 'announcement',
+	ALLIANCE_MOTION = 'alliance_motion',
+	TECHNICAL_PIPS = 'technical PIPs',
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:24]: should be snake_case

File: src/global/proposalType.ts [hunk 15]
```diff
@@ -1,262 +1,292 @@
...
+	case 'AllianceMotion':
+		return 'alliance_motion';
+	case 'Announcement':
+		return 'announcement';
+	case 'Community':
+		return 'community PIPs';
+	case 'TechnicalCommittee':
+		return 'technical PIPs';
+	case 'UpgradeCommitte':
+		return 'upgrade PIPs';
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:127]: should be snake case

File: src/global/proposalType.ts [hunk 16]
```diff
@@ -1,262 +1,292 @@
...
+	case 'tech':
+		return ProposalType.TECH_COMMITTEE_PROPOSALS;
+	case 'tip':
+		return ProposalType.TIPS;
+	case 'treasury':
+		return ProposalType.TREASURY_PROPOSALS;
+	}
+}
+
+export const proposalTypes = ['democracy_proposals', 'tech_committee_proposals', 'treasury_proposals', 'referendums', 'fellowship_referendums', 'council_motions', 'bounties', 'tips', 'child_bounties', 'open_gov', 'referendums_v2', 'alliance_motion', 'announcement','technical PIPs','community PIPs','upgrade PIPs'];
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:230]: snake case

File: src/global/proposalType.ts [hunk 17]
```diff
@@ -1,262 +1,292 @@
...
+export const offChainProposalTypes = ['discussions', 'grants'];
+
+export const checkIsOnChainPost = (proposalType: string) => {
+	return !offChainProposalTypes.includes(proposalType);
+};
+
+export const gov1ProposalTypes = ['DemocracyProposal', 'TechCommitteeProposal', 'TreasuryProposal', 'Referendum', 'CouncilMotion', 'Bounty', 'Tip', 'ChildBounty'];
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:237]: please check if Pips need to be included here

File: src/global/proposalType.ts [hunk 18]
```diff
@@ -1,262 +1,292 @@
...
+	case ProposalType.REFERENDUMS:
+		return 'referendum';
+	case ProposalType.OPEN_GOV:
+		return 'referendumV2';
+	case ProposalType.TIPS:
+		return 'tip';
+	case ProposalType.TECHNICAL_PIPS:
+		return 'technical committee';
+	case ProposalType.UPGRADE_PIPS:
+		return 'technical committee';
```
Comment (alphainfinitus) [inline src/global/proposalType.ts:157]: should be upgrade

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments

File: pages/api/v1/auth/actions/editPost.ts [hunk 19]
```diff
@@ -100,6 +103,8 @@ const handler: NextApiHandler<IEditPostResponse | MessageType> = async (req, res
 				variables
 			});
 
+			console.log(postRes);
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/editPost.ts:106]: please remove console log

File: pages/api/v1/auth/actions/editPost.ts [hunk 20]
```diff
@@ -100,6 +103,8 @@ const handler: NextApiHandler<IEditPostResponse | MessageType> = async (req, res
...
+			console.Console;
```
Comment (alphainfinitus) [inline pages/api/v1/auth/actions/editPost.ts:107]: please remove this line

**Pull request merged**